botorch/optim/utils.py

```diff
@@ -153,7 +153,11 @@ def _expand_bounds(
             ebounds = bounds.view(1, -1)
         else:
             ebounds = bounds
-        if ebounds.shape[1] != X.shape[-1]:
+        if (
+            ebounds.shape[-1] != X.shape[-1]
+            or ebounds.shape[-2] not in [1, X.shape[-2]]
+            or (ebounds.dim() > 2 and ebounds.shape[:-2] != X.shape[:-2])
```

Comment:
So this requires bounds to be defined with explicit t-batch dim. An alternative would be to actually fully support broadcasting - would that make sense her? I wrote some utilities for this a while ago that may be helpful here: https://github.com/cornellius-gp/gpytorch/blob/master/gpytorch/utils/broadcasting.py#L6

Reply:
I looked into the behavior of `Tensor.max()` (or `min()`), and it is safe as long as `bounds` is broadcastable to `X`. It matches the expected behavior as well, so that should work here. I will look into a simpler condition for this.

Reply:
You may be able to just use the broadcasting utils directly? Or I guess that doesn't cover the case when `bounds` has explicit t-batch dims but `X` doesn't?